builder/includes/associations.form.inc

```diff
@@ -298,3 +295,151 @@ function xml_form_builder_enable_association($form_name, $id) {
+        '#value' => $form_name,
+      ),
+      'title_field' => array(
+        '#type' => 'select',
+        '#title' => t('Title Field'),
+        '#description' => t("The form field for the object's label."),
+        '#prefix' => '<div id="ahah-wrapper">',
+        '#suffix' => '</div>',
+        '#options' => $options,
+        '#default_value' => $selected,
+      ),
+      'id' => array(
+        '#type' => 'value',
+        '#value' => $id,
+      ),
+    ),
+  );
+  if (!variable_get('xml_form_builder_use_default_dc_xslts', FALSE)) {
+    $form['fieldset']['transform'] = array(
+      '#type' => 'select',
+      '#title' => t('XSL Transform'),
+      '#options' => xml_form_builder_get_transforms(),
+      '#default_value' => $association['transform'],
+      '#description' => t("An XSL transform for setting the Fedora object's Dublin Core metadata datastream."),
+    );
+    $form['fieldset']['self_transform'] = array(
+      '#type' => 'select',
+      '#title' => t('Self XSL Transform'),
+      '#options' => xml_form_builder_get_self_transforms(),
+      '#default_value' => $association['self_transform'],
+      '#description' => t('An optional transformation applied to form data prior to ingestion.'),
+    );
+  }
+  $form['fieldset']['remove_transform'] = array(
+    '#title' => t('Remove existing template'),
+    '#type' => 'checkbox',
+    '#access' => $association['template'],
+  );
+  $title = $association['template'] ? t('Replace Current Template Document') : t('Upload Template Document');
+  $form['fieldset']['file'] = array(
+    '#type' => 'file',
+    '#title' => filter_xss($title),
+    '#description' => t('A sample metadata file used to prepopulate the form on ingest.'),
+    '#states' => array(
+      'visible' => array(
+        ':input[name="remove_transform"]' => array('checked' => FALSE),
+      ),
+    ),
+  );
+  $form['fieldset']['submit'] = array(
+    '#type' => 'submit',
+    '#value' => t('Edit Association'),
+  );
+  return $form;
+}
+
+/**
+ * Default submit handler for the associations edit form.
+ *
+ * @param array $form
+ *   The drupal form.
+ * @param array $form_state
+ *   The drupal form state.
+ */
+function xml_form_builder_edit_association_form_submit(array $form, array &$form_state) {
+  module_load_include('inc', 'xml_form_builder', 'includes/associations');
+  $id = $form_state['values']['id'];
+  $form_name = $form_state['values']['form_name'];
+  $template = '';
+  $file_uploaded = $_FILES['files']['error']['file'] == 0;
+  if ($file_uploaded & !$form_state['values']['remove_transform']) {
+    $definition = new DOMDocument();
+    $definition->load($_FILES['files']['tmp_name']['file']);
+    $template = $definition->saveXML();
+  }
+  $updates = array(
+    'content_model' => $form_state['values']['content_model'],
+    'dsid' => $form_state['values']['dsid'],
+    'title_field' => $form_state['values']['title_field'],
+    'transform' => $form_state['values']['transform'],
```

Comment:
When [`xml_form_builder_use_default_dc_xslts`](https://github.com/Islandora/islandora_xml_forms/pull/230/files#diff-9738e10d8743356fb7e4b0590e27e48eR366) is selected these form fields aren't rendered and are undefined on submit. Imagine we wouldn't want to nuke these values on submission so maybe render them and use `#access` to not show them to the user?